src/Microsoft.ML.Transforms/Text/TextNormalizerTransform.cs

```diff
@@ -92,294 +92,498 @@ private static VersionInfo GetVersionInfo()
 
-                if (ch != src.Span[i])
-                {
-                    buffer.AppendSpan(span.Slice(min, i - min)).Append(ch);
-                    min = i + 1;
+                    if (ch != src.Span[i])
+                    {
+                        buffer.AppendSpan(span.Slice(min, i - min)).Append(ch);
+                        min = i + 1;
+                    }
+
+                    i++;
                 }
 
-                i++;
+                Host.Assert(i == src.Length);
+                int len = i - min;
+                if (min == 0)
+                {
+                    Host.Assert(src.Length == len);
+                    dst = src;
+                }
+                else
+                {
+                    buffer.AppendSpan(span.Slice(min, len));
+                    dst = buffer.ToString().AsMemory();
+                }
             }
 
-            Host.Assert(i == src.Length);
-            int len = i - min;
-            if (min == 0)
+            /// <summary>
+            /// Whether a character is a combining diacritic character or not.
+            /// Combining diacritic characters are the set of diacritics intended to modify other characters.
+            /// The list is provided by Office NL team.
+            /// </summary>
+            private bool IsCombiningDiacritic(char ch)
             {
-                Host.Assert(src.Length == len);
-                dst = src;
-            }
-            else
-            {
-                buffer.AppendSpan(span.Slice(min, len));
-                dst = buffer.ToString().AsMemory();
+                if (ch < 0x0300 || ch > 0x0670)
+                    return false;
+
+                // Basic combining diacritics
+                return ch >= 0x0300 && ch <= 0x036F ||
+
+                    // Hebrew combining diacritics
+                    ch >= 0x0591 && ch <= 0x05BD || ch == 0x05C1 || ch == 0x05C2 || ch == 0x05C4 ||
+                    ch == 0x05C5 || ch == 0x05C7 ||
+
+                    // Arabic combining diacritics
+                    ch >= 0x0610 && ch <= 0x0615 || ch >= 0x064C && ch <= 0x065E || ch == 0x0670;
             }
         }
+    }
 
+    public sealed class TextNormalizerEstimator : TrivialEstimator<TextNormalizerTransform>
+    {
         /// <summary>
-        /// Whether a character is a combining diacritic character or not.
-        /// Combining diacritic characters are the set of diacritics intended to modify other characters.
-        /// The list is provided by Office NL team.
+        /// Case normalization mode of text. This enumeration is serialized.
         /// </summary>
-        private bool IsCombiningDiacritic(char ch)
+        public enum CaseNormalizationMode
         {
-            if (ch < 0x0300 || ch > 0x0670)
-                return false;
+            Lower = 0,
+            Upper = 1,
+            None = 2
+        }
+
+        public static class Defaults
```

Comment:
>public [](start = 8, length = 6)

Could this be internal, or private instead of public? The user should not need it.